Modules/Scripted/SegmentStatistics/SegmentStatisticsPlugins/ScalarVolumeSegmentStatisticsPlugin.py

```diff
@@ -35,6 +37,40 @@ def computeStatistics(self, segmentID):
       # Input grayscale node does not contain valid image data
       return {}
 
+    voxelsInSegment = self.getVoxelsInSegment(segmentationNode, segmentID, grayscaleNode)
+
+    # Only compute stats using voxels within the display node thresholds for the volume
+    threshold_min = grayscaleNode.GetDisplayNode().GetLowerThreshold()
+    threshold_max = grayscaleNode.GetDisplayNode().GetUpperThreshold()
+    arrayThresholded = voxelsInSegment[voxelsInSegment >= threshold_min]
+    arrayThresholded = arrayThresholded[arrayThresholded <= threshold_max]
```

Comment:
You should maintain current functionality of being able to calculate statistics no matter if the voxel is within the threshold or not. If you want the option of also applying threshold to ignore those voxels in statistics calculations, I would suggest to create another parameter setting. This might exist inside the specific plugin parameters section, or in the Advanced SegmentStatistics module if applicable to other plugins.